bundle/Cache/PersistenceCachePurger.php

```diff
@@ -73,13 +76,23 @@ public function __construct(
     }
 
     /**
-     * Clear all persistence cache.
+     * Clear all persistence cache if that is allowed by config.
      *
      * In legacy kernel used when user presses clear all cache button in admin interface.
+     */
+    public function all()
+    {
+        if ($this->container->getParameter('ezpublish_legacy.clear_all_spi_cache_from_legacy')) {
+            $this->flushSpiCache();
+        }
+    }
+
+    /**
+     * Clear all persistence cache.
      *
      * Sets a internal flag 'allCleared' to avoid clearing cache several times
      */
-    public function all()
+    protected function flushSpiCache()
```

Comment:
We don't need this method to be exposed at all.

```suggestion
    private function flushSpiCache()
```